## Ana Ganfornina

In [137]:
import pandas as pd

In [138]:
data =  pd.read_csv(r"./dataSet/kiva_loans.csv")
original_data = data

In [139]:
data

,id,funded_amount,loan_amount,activity,sector,use,country_code,country,region,currency,partner_id,posted_time,disbursed_time,funded_time,term_in_months,lender_count,tags,borrower_genders,repayment_interval,date
0,653051,300.0,300.0,Fruits & Vegetables,Food,"To buy seasonal, fresh fruits to sell.",PK,Pakistan,Lahore,PKR,247.0,2014-01-01 06:12:39+00:00,2013-12-17 08:00:00+00:00,2014-01-02 10:06:32+00:00,12.0,12,NaN,female,irregular,2014-01-01
1,653053,575.0,575.0,Rickshaw,Transportation,to repair and maintain the auto rickshaw used ...,PK,Pakistan,Lahore,PKR,247.0,2014-01-01 06:51:08+00:00,2013-12-17 08:00:00+00:00,2014-01-02 09:17:23+00:00,11.0,14,NaN,"female, female",irregular,2014-01-01
2,653068,150.0,150.0,Transportation,Transportation,To repair their old cycle-van and buy another ...,IN,India,Maynaguri,INR,334.0,2014-01-01 09:58:07+00:00,2013-12-17 08:00:00+00:00,2014-01-01 16:01:36+00:00,43.0,6,"user_favorite, user_favorite",female,bullet,2014-01-01
3,653063,200.0,200.0,Embroidery,Arts,to purchase an embroidery machine and a variet...,PK,Pakistan,Lahore,PKR,247.0,2014-01-01 08:03:11+00:00,2013-12-24 08:00:00+00:00,2014-01-01 13:00:00+00:00,11.0,8,NaN,female,irregular,2014-01-01
4,653084,400.0,400.0,Milk Sales,Food,to purchase one buffalo.,PK,Pakistan,Abdul Hakeem,PKR,245.0,2014-01-01 11:53:19+00:00,2013-12-17 08:00:00+00:00,2014-01-01 19:18:51+00:00,14.0,16,NaN,female,monthly,2014-01-01
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
671200,1340323,0.0,25.0,Livestock,Agriculture,"[True, u'para compara: cemento, arenya y ladri...",PY,Paraguay,Concepción,USD,58.0,2017-07-25 16:55:34+00:00,2017-07-25 07:00:00+00:00,NaN,13.0,0,NaN,female,monthly,2017-07-25
671201,1340316,25.0,25.0,Livestock,Agriculture,"[True, u'to start a turducken farm.'] - this l...",KE,Kenya,NaN,KES,138.0,2017-07-25 06:14:08+00:00,2017-07-24 07:00:00+00:00,2017-07-26 02:09:43+00:00,13.0,1,NaN,female,monthly,2017-07-25
671202,1340334,0.0,25.0,Games,Entertainment,NaN,KE,Kenya,NaN,KES,138.0,2017-07-26 00:02:07+00:00,2017-07-25 07:00:00+00:00,NaN,13.0,0,NaN,NaN,monthly,2017-07-26
671203,1340338,0.0,25.0,Livestock,Agriculture,"[True, u'to start a turducken farm.'] - this l...",KE,Kenya,NaN,KES,138.0,2017-07-26 06:12:55+00:00,2017-07-25 07:00:00+00:00,NaN,13.0,0,NaN,female,monthly,2017-07-26


In [140]:
original_data.info()

<class 'pandas.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 20 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   loan_amount         671205 non-null  float64
 3   activity            671205 non-null  str    
 4   sector              671205 non-null  str    
 5   use                 666973 non-null  str    
 6   country_code        671197 non-null  str    
 7   country             671205 non-null  str    
 8   region              614405 non-null  str    
 9   currency            671205 non-null  str    
 10  partner_id          657698 non-null  float64
 11  posted_time         671205 non-null  str    
 12  disbursed_time      668809 non-null  str    
 13  funded_time         622874 non-null  str    
 14  term_in_months      671205 non-null  float64
 15  lender_count        671205 non-null  int64  


In [141]:
original_data.shape

(671205, 20)

In [142]:
data.describe() # parecido al estatistics de excel


,id,funded_amount,loan_amount,partner_id,term_in_months,lender_count
count,6.712050e+05,671205.000000,671205.000000,657698.000000,671205.000000,671205.000000
mean,9.932486e+05,785.995061,842.397107,178.199616,13.739022,20.590922
std,1.966113e+05,1130.398941,1198.660073,94.247581,8.598919,28.459551
min,6.530470e+05,0.000000,25.000000,9.000000,1.000000,0.000000
25%,8.230720e+05,250.000000,275.000000,126.000000,8.000000,7.000000
50%,9.927800e+05,450.000000,500.000000,145.000000,13.000000,13.000000
75%,1.163653e+06,900.000000,1000.000000,204.000000,14.000000,24.000000
max,1.340339e+06,100000.000000,100000.000000,536.000000,158.000000,2986.000000


## Limpieza de datos

### Limpieza de fechas

- Pasar de tipo str a datetime las fechas

In [143]:
data_type = data.loc[:,['posted_time', 'disbursed_time', 'funded_time', 'date']]
data_type.info()

<class 'pandas.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 4 columns):
 #   Column          Non-Null Count   Dtype
---  ------          --------------   -----
 0   posted_time     671205 non-null  str  
 1   disbursed_time  668809 non-null  str  
 2   funded_time     622874 non-null  str  
 3   date            671205 non-null  str  
dtypes: str(4)
memory usage: 20.5 MB


In [144]:
colum_date = ['posted_time', 'disbursed_time', 'funded_time', 'date']
for column in colum_date:
    data[column] = pd.to_datetime(data[column]).dt.date #Por cada columna cambia a tipo fecha

data_type = data.loc[:,['posted_time', 'disbursed_time', 'funded_time', 'date']]
data_type.info()

<class 'pandas.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 4 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   posted_time     671205 non-null  object
 1   disbursed_time  668809 non-null  object
 2   funded_time     622874 non-null  object
 3   date            671205 non-null  object
dtypes: object(4)
memory usage: 20.5+ MB


Debido a que no usaremos en nuestro analisis las horas de las fechas, las formateamos, quedandonos tan solo con el dia, mes y año. Como tipo de dato aparecen como si fueran object, pero internamente siguen siendo datatime.date, lo que permite seguir haciendo filtrado

### Comprobación de nulos

In [145]:
data.isnull()

,id,funded_amount,loan_amount,activity,sector,use,country_code,country,region,currency,partner_id,posted_time,disbursed_time,funded_time,term_in_months,lender_count,tags,borrower_genders,repayment_interval,date
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
671200,False,False,False,False,False,False,False,False,False,False,False,False,False,True,False,False,True,False,False,False
671201,False,False,False,False,False,False,False,False,True,False,False,False,False,False,False,False,True,False,False,False
671202,False,False,False,False,False,True,False,False,True,False,False,False,False,True,False,False,True,True,False,False
671203,False,False,False,False,False,False,False,False,True,False,False,False,False,True,False,False,True,False,False,False


In [146]:
data.isnull().mean() * 100

id                     0.000000
funded_amount          0.000000
loan_amount            0.000000
activity               0.000000
sector                 0.000000
use                    0.630508
country_code           0.001192
country                0.000000
region                 8.462392
currency               0.000000
partner_id             2.012351
posted_time            0.000000
disbursed_time         0.356970
funded_time            7.200632
term_in_months         0.000000
lender_count           0.000000
tags                  25.538546
borrower_genders       0.628869
repayment_interval     0.000000
date                   0.000000
dtype: float64

In [147]:
data = data.drop(columns=['tags']) #Eliminamos tags debido a que tiene un 25% de nulos
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 19 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   loan_amount         671205 non-null  float64
 3   activity            671205 non-null  str    
 4   sector              671205 non-null  str    
 5   use                 666973 non-null  str    
 6   country_code        671197 non-null  str    
 7   country             671205 non-null  str    
 8   region              614405 non-null  str    
 9   currency            671205 non-null  str    
 10  partner_id          657698 non-null  float64
 11  posted_time         671205 non-null  object 
 12  disbursed_time      668809 non-null  object 
 13  funded_time         622874 non-null  object 
 14  term_in_months      671205 non-null  float64
 15  lender_count        671205 non-null  int64  


- Sustitución de nulos por "No especificado" en caso de string y por la media numerica de la columna en caso de número

In [148]:

#Seleccionar los nombres de las columnas que son texto
columns_str = data.select_dtypes(str).columns
print(columns_str)

for column in columns_str:
    data[column] = data[column].fillna("No especificado")

# Seleccionamos las que son int o float
columns_num = data.select_dtypes('number').columns

for column in columns_num:
    # Calculamos la mediana de la columna
    mediana = data[column].median()
    # Rellenamos los nulos de esa columna con su mediana
    data[column] = data[column].fillna(mediana)

# Verificamos que ya no quedan nulos
print(data.isnull().sum())




Index(['activity', 'sector', 'use', 'country_code', 'country', 'region',
       'currency', 'borrower_genders', 'repayment_interval'],
      dtype='str')
id                        0
funded_amount             0
loan_amount               0
activity                  0
sector                    0
use                       0
country_code              0
country                   0
region                    0
currency                  0
partner_id                0
posted_time               0
disbursed_time         2396
funded_time           48331
term_in_months            0
lender_count              0
borrower_genders          0
repayment_interval        0
date                      0
dtype: int64


In [149]:
data

,id,funded_amount,loan_amount,activity,sector,use,country_code,country,region,currency,partner_id,posted_time,disbursed_time,funded_time,term_in_months,lender_count,borrower_genders,repayment_interval,date
0,653051,300.0,300.0,Fruits & Vegetables,Food,"To buy seasonal, fresh fruits to sell.",PK,Pakistan,Lahore,PKR,247.0,2014-01-01,2013-12-17,2014-01-02,12.0,12,female,irregular,2014-01-01
1,653053,575.0,575.0,Rickshaw,Transportation,to repair and maintain the auto rickshaw used ...,PK,Pakistan,Lahore,PKR,247.0,2014-01-01,2013-12-17,2014-01-02,11.0,14,"female, female",irregular,2014-01-01
2,653068,150.0,150.0,Transportation,Transportation,To repair their old cycle-van and buy another ...,IN,India,Maynaguri,INR,334.0,2014-01-01,2013-12-17,2014-01-01,43.0,6,female,bullet,2014-01-01
3,653063,200.0,200.0,Embroidery,Arts,to purchase an embroidery machine and a variet...,PK,Pakistan,Lahore,PKR,247.0,2014-01-01,2013-12-24,2014-01-01,11.0,8,female,irregular,2014-01-01
4,653084,400.0,400.0,Milk Sales,Food,to purchase one buffalo.,PK,Pakistan,Abdul Hakeem,PKR,245.0,2014-01-01,2013-12-17,2014-01-01,14.0,16,female,monthly,2014-01-01
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
671200,1340323,0.0,25.0,Livestock,Agriculture,"[True, u'para compara: cemento, arenya y ladri...",PY,Paraguay,Concepción,USD,58.0,2017-07-25,2017-07-25,NaT,13.0,0,female,monthly,2017-07-25
671201,1340316,25.0,25.0,Livestock,Agriculture,"[True, u'to start a turducken farm.'] - this l...",KE,Kenya,No especificado,KES,138.0,2017-07-25,2017-07-24,2017-07-26,13.0,1,female,monthly,2017-07-25
671202,1340334,0.0,25.0,Games,Entertainment,No especificado,KE,Kenya,No especificado,KES,138.0,2017-07-26,2017-07-25,NaT,13.0,0,No especificado,monthly,2017-07-26
671203,1340338,0.0,25.0,Livestock,Agriculture,"[True, u'to start a turducken farm.'] - this l...",KE,Kenya,No especificado,KES,138.0,2017-07-26,2017-07-25,NaT,13.0,0,female,monthly,2017-07-26


En un inicio y debido a que los  2396 valores nulos de disbursed_time y los  48331 de funded_time representan solo el 0.36% y 7.2%  del dataSet respectivamente, pensamos en eliminar dichas filas. Sin embargo nos dimos cuenta que al hacerlo estabamos truncando la información relativa a los prestamos no financiados. Es por ello que decidimos dejarlos ya que no eran una gran cantidad de nulos.    

In [150]:
# data = data.dropna() #borra las filas que tengan valores faltantes
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 19 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   loan_amount         671205 non-null  float64
 3   activity            671205 non-null  str    
 4   sector              671205 non-null  str    
 5   use                 671205 non-null  str    
 6   country_code        671205 non-null  str    
 7   country             671205 non-null  str    
 8   region              671205 non-null  str    
 9   currency            671205 non-null  str    
 10  partner_id          671205 non-null  float64
 11  posted_time         671205 non-null  object 
 12  disbursed_time      668809 non-null  object 
 13  funded_time         622874 non-null  object 
 14  term_in_months      671205 non-null  float64
 15  lender_count        671205 non-null  int64  


In [151]:
print(data.isnull().sum())

id                        0
funded_amount             0
loan_amount               0
activity                  0
sector                    0
use                       0
country_code              0
country                   0
region                    0
currency                  0
partner_id                0
posted_time               0
disbursed_time         2396
funded_time           48331
term_in_months            0
lender_count              0
borrower_genders          0
repayment_interval        0
date                      0
dtype: int64


### Eliminación de duplicados

In [152]:
total_duplicados = data.duplicated().sum() #cuenta el número de duplicados
print(f"Tienes {total_duplicados} filas duplicadas.") #como no hay ninguno, no hacemos nada.

Tienes 0 filas duplicadas.


### Normalización de textos

In [153]:
for column in data.columns:
    if data[column].dtype == 'str':
        data[column] = data[column].str.strip().str.capitalize() #Eliminamos espacios en blanco y ponemos la primera letra en mayusculas



### Normalización de monedas

- Eliminación de country code debido a redundancia con la columna de country y conversión de currency a dolares para poder eliminar la misma columna de country

In [154]:
data = data.drop(columns=['country_code']) #Eliminación de columna contry_code

In [155]:
# import requests

# Obtenemos tasas de cambio
# response = requests.get('https://api.exchangerate-api.com/v4/latest/USD')
# rates_usd_base = response.json()['rates']

# Creamos un diccionario vacio al cual decimos que recorra el de rates_usd_base para invertir los valores
# rates_to_usd = {}
# for currency, rate in rates_usd_base.items():
    # rates_to_usd[currency] = 1 / rate

# data['currency'] = data['currency'].str.upper() #normalizamos a mayusculas para que coincidan con los datos de conversion de la API

# def get_rate(currency_code):
  #   return rates_to_usd.get(currency_code, None) # En caso de no encontrar la moneda devolveremos None


#Creamos dos nuevas columnas para meter estos nuevos datos
# data['funded_amount_usd'] = data['funded_amount'] * data['currency'].apply(get_rate)
# data['loan_amount_usd'] = data['loan_amount'] * data['currency'].apply(get_rate)



In [156]:
#Eliminar  funded_amount y loan_amount puesto que ya no son necesarias y podemos calcularas en base a funded_amount_usd y la tasa de conversión
# data = data.drop(columns=['funded_amount','loan_amount'])

Después de releer la documentación de Kiva nos percatamos de que las cifras de funded_amount y loan_amout ya estaban en USD por lo que el código anterior no tendría mucho sentido, pero deja abierta una via de investigación para evaluar la relevancia de los prestamos según el pais.

### Eliminación de columnas ID y partner ID 

No es muy significativa para el analisis y en lugar de convertirlos a numeros las eliminamos

In [157]:
data = data.drop(columns=['id','partner_id'])
print (data.columns)

Index(['funded_amount', 'loan_amount', 'activity', 'sector', 'use', 'country',
       'region', 'currency', 'posted_time', 'disbursed_time', 'funded_time',
       'term_in_months', 'lender_count', 'borrower_genders',
       'repayment_interval', 'date'],
      dtype='str')


### limpieza numerica

Redondeamos los datos numericos a dos decimales al tratarse de microprestamos, pues ambos decimales pueden ser de gran utilidad.
Sin embargo redondearemos a enteros en el caso de los meses y los solicitantes del prestamos para una mayor claridad

In [158]:
columns_num = data.select_dtypes('number').columns
data[columns_num] = data[columns_num].round(2) 
data[['term_in_months', 'lender_count']] = data[['term_in_months', 'lender_count']].astype(int)

## Analisis exploratorio

1º LINEA DE ANALISIS: EVOLUCION MENSUAL Y ESTACIONALIDAD - EL FACTOR TIEMPO (date)

In [159]:
#Aseguramos formato fecha
data['date'] = pd.to_datetime(data['date'])
#Extraemos año y mes para analisis de estacionalidad
data['year'] = data['date'].dt.year
data['month'] = data['date'].dt.month

In [160]:
# Agrupamos por mes real
grupos_mensuales = data['date'].dt.to_period('M')
evolucion_mensual = data.groupby(grupos_mensuales).agg(
    conteo_prestamos=('funded_amount', 'count'),
    monto_total=('funded_amount', 'sum'),
    monto_promedio=('funded_amount', 'mean')
)

In [161]:
# Calculamos el crecimiento porcentual mes a mes
evolucion_mensual['crecimiento_vol_pct'] = evolucion_mensual['conteo_prestamos'].pct_change() * 100
# Mostramos resultado
print(evolucion_mensual.head(12))

         conteo_prestamos  monto_total  monto_promedio  crecimiento_vol_pct
date                                                                       
2014-01             11735   10345250.0      881.572220                  NaN
2014-02             13697   12381025.0      903.922392            16.719216
2014-03             13627   12023630.0      882.338739            -0.511061
2014-04             13509   11789590.0      872.721149            -0.865928
2014-05             14108   11610830.0      822.996172             4.434081
2014-06             14301   12260855.0      857.342494             1.368018
2014-07             15241   13062715.0      857.077292             6.572967
2014-08             14722   11963140.0      812.602907            -3.405288
2014-09             15405   12399960.0      804.930867             4.639315
2014-10             16377   13046885.0      796.659034             6.309640
2014-11             16484   13212440.0      801.531182             0.653355
2014-12     

- CONCLUSION: 
El volumen de préstamos muestra una tendencia de crecimiento anual con picos máximos en el último trimestre (octubre-noviembre). Sin embargo, existe una estacionalidad negativa marcada en diciembre, donde el volumen cae drásticamente. Además, se observa que en los meses de mayor actividad, el monto promedio por préstamo tiende a disminuir, sugiriendo una mayor diversificación hacia microcréditos de menor escala.

In [162]:
#Aquí agrupamos todos los 'meses'..etc
estacionalidad = data.groupby('month').agg(
    promedio_mensual_prestamos=('funded_amount', 'count'),
    promedio_monto_solicitado=('funded_amount', 'mean')
).sort_values('month')
print("--- EVOLUCIÓN TEMPORAL (Primeras 10 filas) ---")
print(evolucion_mensual.head(10))

print("\n--- PATRÓN POR MES (ESTACIONALIDAD) ---")
print(estacionalidad)

--- EVOLUCIÓN TEMPORAL (Primeras 10 filas) ---
         conteo_prestamos  monto_total  monto_promedio  crecimiento_vol_pct
date                                                                       
2014-01             11735   10345250.0      881.572220                  NaN
2014-02             13697   12381025.0      903.922392            16.719216
2014-03             13627   12023630.0      882.338739            -0.511061
2014-04             13509   11789590.0      872.721149            -0.865928
2014-05             14108   11610830.0      822.996172             4.434081
2014-06             14301   12260855.0      857.342494             1.368018
2014-07             15241   13062715.0      857.077292             6.572967
2014-08             14722   11963140.0      812.602907            -3.405288
2014-09             15405   12399960.0      804.930867             4.639315
2014-10             16377   13046885.0      796.659034             6.309640

--- PATRÓN POR MES (ESTACIONALIDAD) ---


- CONCLUSION: 
El mes con  mayor volumen de prestamos es el mes de marzo con 67.583 préstamos. Podemos observar que hay un aumento de enero hasta marzo. Puede que venga motivado por el inicio de ciclos productivos o escolares en muchas regiones del mundo. A partir del mes de julio, el volumen cae significativamente y se mantiene en un nivel bajo y estable hasta final de año. Podemos observar que enero y febrero tienen los promedios de monto solicitados más altos a pesar de no ser los meses con más creéitos concedidos.


2º LINEA DE ANALISIS: CUALES SON LOS PAISES EN LOS QUE MAS CREDITOS SE DAN

In [163]:
data.describe()

,funded_amount,loan_amount,term_in_months,lender_count,date,year,month
count,671205.000000,671205.000000,671205.000000,671205.000000,671205,671205.000000,671205.000000
mean,785.995061,842.397107,13.739022,20.590922,2015-11-09 22:40:34,2015.385584,6.165738
min,0.000000,25.000000,1.000000,0.000000,2014-01-01 00:00:00,2014.000000,1.000000
25%,250.000000,275.000000,8.000000,7.000000,2014-12-17 00:00:00,2014.000000,3.000000
50%,450.000000,500.000000,13.000000,13.000000,2015-11-20 00:00:00,2015.000000,6.000000
75%,900.000000,1000.000000,14.000000,24.000000,2016-10-06 00:00:00,2016.000000,9.000000
max,100000.000000,100000.000000,158.000000,2986.000000,2017-07-26 00:00:00,2017.000000,12.000000
std,1130.398941,1198.660073,8.598919,28.459551,NaN,1.052327,3.374374


In [164]:
data.loc[:,['country','funded_amount','loan_amount']]

,country,funded_amount,loan_amount
0,Pakistan,300.0,300.0
1,Pakistan,575.0,575.0
2,India,150.0,150.0
3,Pakistan,200.0,200.0
4,Pakistan,400.0,400.0
...,...,...,...
671200,Paraguay,0.0,25.0
671201,Kenya,25.0,25.0
671202,Kenya,0.0,25.0
671203,Kenya,0.0,25.0


In [165]:
country_top = data.groupby('country')['loan_amount'].sum().sort_values(ascending=False) #buscamos en el dataset en el la columna country y mostramos los valores de forma desdendente
country_top.head(10).to_frame()



,loan_amount
country,
Philippines,55342225.0
Kenya,34534300.0
United states,31496375.0
Peru,30867400.0
Paraguay,29629075.0
El salvador,26603650.0
Cambodia,19872700.0
Bolivia,19843250.0
Rwanda,16646300.0


- CONLUSION: Estos son los 10 paises donde la suma de sus totales nos da una mayor cantidad de dinero solicitada

3º LINEA DE ANALISIS: QUE PRESTAMOS RECIBEN MAS DINERO ¿LOS DE MAYOR NUMERO DE SOLICITANTES O LOS DE MENOR?

In [166]:
lenders = data.groupby('lender_count')['loan_amount'].max().sort_values(ascending=False)
lenders.head(5).to_frame()

,loan_amount
lender_count,
2986,100000.0
1170,50000.0
1441,50000.0
1410,50000.0
1402,50000.0


- CONCLUSION: Como conclusión de la tabla anterior deducimos que a mayor numero de personas solicitantes del préstamo mayor es la cuantía solicitada

4º LINEA DE ANALISIS: ¿CUAL ES LA RELACION ENTRE LOS MONTOS? ¿HAY PRESTAMOS QUE SE QUEDAN SIN FINANCIAR?

In [167]:
data.loc[:,['funded_amount','loan_amount']]

,funded_amount,loan_amount
0,300.0,300.0
1,575.0,575.0
2,150.0,150.0
3,200.0,200.0
4,400.0,400.0
...,...,...
671200,0.0,25.0
671201,25.0,25.0
671202,0.0,25.0
671203,0.0,25.0


In [168]:
no_completados = (data['funded_amount'] < data['loan_amount']).sum()

cero_financiacion = (data['funded_amount'] == 0).sum() #Contamos cuantos tienen 0 financiación

print(f"Préstamos con financiación incompleta: {no_completados}")
print(f"Préstamos con CERO financiación: {cero_financiacion}")

Préstamos con financiación incompleta: 48328
Préstamos con CERO financiación: 3383


In [169]:
pct_incompletos = (data['funded_amount'] < data['loan_amount']).mean() * 100

print(f"Porcentaje de prestamos sin completar: {pct_incompletos:.2f}%")

Porcentaje de prestamos sin completar: 7.20%


- CONCLUSION: Debido a que el porcentaje sin 7.2% podemos deducir que existe una gran demanda que la plataforma no logra cubrir.